src/store/search-store.js

```diff
@@ -131,7 +144,19 @@ const actions = ImageService => ({
           commit(IMAGE_NOT_FOUND);
         }
         else {
-          commit(FETCH_IMAGES_ERROR);
+          if (error.response) {
+            if (error.response.status === 500) {
+              commit(FETCH_IMAGES_ERROR, { errorMsg: 'There was a problem with our servers' });
+            }
+            else if (error.response.status === 404) {
+              commit(FETCH_IMAGES_ERROR, { errorMsg: 'No images were found for this image' });
+            }
+            else {
+              commit(FETCH_IMAGES_ERROR, { errorMsg: error.response.message });
+            }
+            return;
+          }
+          commit(FETCH_IMAGES_ERROR, { errorMsg: error.message });
```

Comment:
Could you create a function called maybe `handleImageLoadError` that has this error handling logic and not have this code duplicated?

Reply:
Okay sure. I will do that

Reply:
Can I make the function an action of the store since mutations need to be committed?